.github/workflows/ci.yml

```diff
@@ -23,14 +23,17 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4
+      - name: Install uv
+        uses: astral-sh/setup-uv@v2
+        with:
+          enable-cache: "true"
       - uses: actions/setup-python@v5
         with:
           python-version: "3.12"
```

Comment:
https://github.com/astral-sh/setup-uv?tab=readme-ov-file#do-i-still-need-actionssetup-python-alongside-setup-uv

> This action is modelled as a drop-in replacement for actions/setup-python when using uv. With setup-uv, you can install a specific version of Python using uv python install rather than relying on actions/setup-python.

Reply:
https://docs.astral.sh/uv/guides/integration/github/#setting-up-python 

> Alternatively, the official GitHub setup-python action can be used. This can be faster, because GitHub caches the Python versions alongside the runner.

Installing python can be done through `uv` but it may be slower then `setup-python`. That's why I have chosen for `setup-python`.